src/Events/Admin/Notice/Rest_Api.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ *
+ * @package TEC\Events\Admin\Notice
+ */
+
+namespace TEC\Events\Admin\Notice;
+
+use Tribe\Events\Views\V2\Rest_Endpoint as V2;
+
+/**
+ * Class Rest_Api
+ *
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ */
+class Rest_Api {
+
+	/**
+	 * Notice Slug on the user options
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $slug = 'events-rest-api-notice';
+
+	/**
+	 * Blocked endpoint.
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $blocked_endpoint = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since
+	 *
+	 * @return void
+	 */
+	public function hook(): void {
+		$slug = $this->slug;
+
+		tribe_notice(
+			$slug,
+			[ $this, 'notice' ],
+			[
+				'type'    => 'error',
+				'dismiss' => 1,
+				'wrap'    => 'p',
+			],
+			[ $this, 'should_display' ]
+		);
+	}
+
+	/**
+	 * Checks if we are in a TEC page or in the main Dashboard.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function should_display(): bool {
+		global $pagenow;
+
+		if ( tribe( 'admin.helpers' )->is_screen() || 'index.php' === $pagenow ) {
+			return $this->is_rest_api_blocked() ? true : false;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if our endpoints are accessible.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function is_rest_api_blocked(): bool {
+
+		$v1_api    = new \Tribe__Events__REST__V1__Main();
+		$event_api = get_rest_url( null, $v1_api->get_events_route_namespace() );
+		$response  = wp_remote_get( $event_api );
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			$this->blocked_endpoint = $event_api;
+			return true;
+		}
+
+		$views_api = get_rest_url( null, V2::ROOT_NAMESPACE );
+		$response  = wp_remote_get( $views_api );
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			$this->blocked_endpoint = $views_api;
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * HTML for the notice when we have blocked REST API endpoints.
+	 *
+	 * @since
+	 *
+	 * @return false|string
+	 */
+	public function notice() {
+		if ( ! current_user_can( 'activate_plugins' ) ) {
+			return false;
+		}
+
+		$text = [];
+		// Translators: %s is the "Warning" word in bold.
+		$text[] = sprintf( __( '%s : The Events Calendar REST API endpoints are not accessible! This may be due to a server configuration or another plugin blocking access to the REST API.', 'the-events-calendar' ), '<strong>' . __( 'Warning', 'the-events-calendar' ) . '</strong>' );
+		$text[] = __( 'Please check with your hosting provider or system administrator to ensure that the below is accessible:', 'the-events-calendar' );
+		$text[] = '<p><a href="' . $this->blocked_endpoint . '" target="_blank">' . $this->blocked_endpoint . '</a></p>';
```

Comment:
Why is this text the only piece wrapped in a paragraph?
Instead of the implode below, wrap them in `<p>`

Reply:
Maybe 
```
		$text[] = sprintf(
			__( %1$sPlease check with your hosting provider or system administrator to ensure that %2$s is accessible.%3$s', 'the-events-calendar' ),
			'<p>',
			'<a href="' . $this->blocked_endpoint . '" target="_blank">' . $this->blocked_endpoint . '</a>','
			'</p>'
		);
```

And then just 
`return implode( '', $text );`

Reply:
I would have never use that implode approach for a translation string. This was [copied from here](https://github.com/the-events-calendar/the-events-calendar/blob/master/src/Tribe/Admin/Notice/Timezones.php) and as we discussed in our last meeting, I wasn't aware of which parts of the plugin are legacy code and which to follow as a standard.

As for the actual text, this is an example I came up with and I was waiting for proper specs.